### DISCUSSION TITLE
Welcome to bat Discussions!

### DISCUSSION THREAD

**Discussion by sharkdp** [sharkdp/bat#2199] [Announcements]
Welcome,

We as maintainers decided that we will give GitHub Discussions a try. Please feel free to use this space to
  * Ask questions about `bat`.
  * Discuss potential new features.
  * Share ideas.
  * Engage with other community members.

**Comment by cpAtor** [comment]
# Help to quickly ramp up and contribute code to bat.

Hi,

I have 3 years web development experience and started learning basics of rust. 
bat looks like an interesting project, and I want to contribute to bat as I am learning Rust. I am trying to pick up some "good first issue's". But I am not able to understand how the control flows in the project.
Are there any resources for developers to ramp up with bat quickly? Or is there some one already working on bat I can contact to better understand the source code? please point me in the right direction.

Thanks in advance for the help.

**Reply by Enselic** [reply]
The entry point to the `bat` binary in the code is `fn main()` in `src/bin/bat/main.rs`. That's where the flow starts. So just follow along from there :) It is also worth mentioning that the output of bat is redirected to a pager (read: `less`) by default. To make the flow of the code easier to understand I would recommend you to disable the pager by passing `--pager=never` to `bat`.

**Reply by Enselic** [reply]
I would also highly recommend VS Code + rust-analyzer so that you can do things like Go To Definition (F12) to very easily navigate around in the code base.

**Reply by interdisziplina** [reply]
@Enselic This is also possible with any LSP (like coc) and not limited to a graphical editor like VS Code.
